master/internal/proxy/proxy.go

```diff
@@ -51,55 +27,60 @@ type ProxyHTTPAuth func(echo.Context) (done bool, err error)
 
 // Proxy is an actor that proxies requests to registered services.
 type Proxy struct {
-	lock     sync.RWMutex
-	services map[string]*Service
-
 	HTTPAuth ProxyHTTPAuth
+	Services map[string]*Service
+	Syslog   *logrus.Entry
 }
 
-// Receive implements the actor.Actor interface.
-func (p *Proxy) Receive(ctx *actor.Context) error {
-	switch msg := ctx.Message().(type) {
-	case actor.PreStart:
-		p.services = make(map[string]*Service)
-	case Register:
-		if msg.ServiceID == "" {
-			return nil
-		}
-		p.lock.Lock()
-		defer p.lock.Unlock()
-		ctx.Log().Infof("registering service: %s (%v)", msg.ServiceID, msg.URL)
-		p.services[msg.ServiceID] = &Service{
-			URL:                  msg.URL,
-			LastRequested:        time.Now(),
-			ProxyTCP:             msg.ProxyTCP,
-			AllowUnauthenticated: msg.Unauthenticated,
-		}
+// TheOneProxy is the global proxy singleton.
+var TheOneProxy *Proxy
 
-		if ctx.ExpectingResponse() {
-			ctx.Respond(nil)
-		}
-	case Unregister:
-		p.lock.Lock()
-		defer p.lock.Unlock()
-		delete(p.services, msg.ServiceID)
-	case NewProxyHandler:
-		ctx.Respond(p.newProxyHandler(msg.ServiceID))
-	case GetSummary:
-		ctx.Respond(p.getSummary())
-	case actor.PostStop:
-		p.lock.Lock()
-		defer p.lock.Unlock()
-		// Erase all services from the proxy in case any handlers are still active.
-		p.services = nil
+// InitProxy initializes the global proxy.
+func InitProxy(httpAuth ProxyHTTPAuth) {
+	if TheOneProxy != nil {
+		logrus.Warn(
+			"detected re-initialization of Proxy that should never occur outside of tests",
+		)
+	}
+	TheOneProxy = &Proxy{
+		HTTPAuth: httpAuth,
+		Services: make(map[string]*Service),
+		Syslog:   logrus.WithTime(time.Now()), // NIT: How else to initialize syslog?
 	}
-	return nil
 }
 
-func (p *Proxy) getService(serviceName string) *Service {
-	p.lock.Lock()
-	defer p.lock.Unlock()
-	service := p.services[serviceName]
+// Register registers the service name with the associated target URL. All requests with the
+// format ".../:service-name/*" are forwarded to the service via the target URL.
+func Register(serviceID string, url *url.URL, proxyTCP bool, unauth bool) {
+	if serviceID == "" {
+		return
+	}
+	// NIT CAROLINA: When to fail?
+	TheOneProxy.Syslog.Infof("registering service: %s (%v)", serviceID, url)
+	TheOneProxy.Services[serviceID] = &Service{
+		URL:                  url,
+		LastRequested:        time.Now(),
+		ProxyTCP:             proxyTCP,
+		AllowUnauthenticated: unauth,
+	}
+	return
+}
+
+// Unregister removes the service from the proxy. All future requests until the service name is
+// registered again will be responded with a 404 response. If the service is not registered with
+// the proxy, the message is ignored.
+func Unregister(serviceID string) {
+	delete(TheOneProxy.Services, serviceID)
+}
+
+// ClearProxy erases all services from the proxy in case any handlers are still active.
+func ClearProxy() {
+	TheOneProxy.Services = nil
```

Comment:
you probably want to set it to a non-nil map instead, but i may clear this API and just reinit the map